### PR TITLE
Allow intersection type for resource signature params and return type

### PR DESCRIPTION
**Pull request by chamil321**
## Checklist
- [x] Linked to an issue
- [x] Updated the changelog
- [x] Added tests
- [ ] ~Updated the spec~

### CODE REVIEW

File: compiler-plugin/src/main/java/io/ballerina/stdlib/http/compiler/HttpInterceptorResourceValidator.java [hunk 1]
```diff
@@ -119,21 +119,23 @@ private static boolean isServiceType(TypeSymbol returnTypeSymbol) {
     }
 
     private static void reportResourceAnnotationNotAllowed(SyntaxNodeAnalysisContext ctx, AnnotationNode node) {
-        HttpCompilerPluginUtil.updateDiagnostic(ctx, node.location(), node.annotReference().toString(),
-                                                HttpDiagnosticCodes.HTTP_125);
+        HttpCompilerPluginUtil.updateDiagnostic(ctx, node.location(), HttpDiagnosticCodes.HTTP_125,
+                                                node.annotReference().toString()
```
Comment (TharmiganK) [inline compiler-plugin/src/main/java/io/ballerina/stdlib/http/compiler/HttpInterceptorResourceValidator.java:123]:
```suggestion
                                                node.annotReference().toString());
```

File: compiler-plugin/src/main/java/io/ballerina/stdlib/http/compiler/HttpInterceptorResourceValidator.java [hunk 2]
```diff
@@ -119,21 +119,23 @@ private static boolean isServiceType(TypeSymbol returnTypeSymbol) {
...
+        );
```
Comment (TharmiganK) [inline compiler-plugin/src/main/java/io/ballerina/stdlib/http/compiler/HttpInterceptorResourceValidator.java:124]:
```suggestion
```

File: native/src/main/java/io/ballerina/stdlib/http/api/HttpDispatcher.java [hunk 3]
```diff
@@ -204,7 +185,7 @@ public static URI getValidatedURI(String uriStr) {
      * This method finds the matching resource for the incoming request.
      *
      * @param servicesRegistry HTTP service registry
-     * @param inboundMessage incoming message.
+     * @param inboundMessage   incoming message.
```
Comment (TharmiganK) [inline native/src/main/java/io/ballerina/stdlib/http/api/HttpDispatcher.java:188]:
```suggestion
     * @param inboundMessage incoming message.
```

File: native/src/main/java/io/ballerina/stdlib/http/api/service/signature/PayloadParam.java [hunk 4]
```diff
@@ -65,4 +99,239 @@ public String getToken() {
...
+                    }
+                    break;
+                case TypeTags.RECORD_TYPE_TAG:
+                    Object recordEntity = getRecordEntity(inRequestEntity, payloadType);
+                    if (this.readonly && recordEntity instanceof BRefValue) {
+                        ((BRefValue) recordEntity).freezeDirect();
+                    }
+                    paramFeed[index++] = recordEntity;
+                    break;
+//            case TypeTags.INTERSECTION_TAG:
```
Comment (TharmiganK) [inline native/src/main/java/io/ballerina/stdlib/http/api/service/signature/PayloadParam.java:238]: Can we remove these commented lines?

**Review by TharmiganK — APPROVED** [review]

File: native/src/main/java/io/ballerina/stdlib/http/api/HttpDispatcher.java (hunk 3 above)
Comment (chamil321) [reply native/src/main/java/io/ballerina/stdlib/http/api/HttpDispatcher.java:188]:
```
    /**
     * This method finds the matching resource for the incoming request.
     *
     * @param servicesRegistry HTTP service registry
     * @param inboundMessage   incoming message.
     * @return matching resource.
     */
     ```
This existing alignment is fine right.

**Pull request merged**